pkg/export/export.go

```diff
@@ -467,19 +488,12 @@ func (e *Exporter) Run(ctx context.Context) error {
 	// Starting index when iterating over shards. This ensures we don't always start at 0 so that
 	// some shards may never be sent in a busy collector.
 	shardOffset := 0
-
-	for {
+	drainShardsAttempt := func(drainCtx context.Context) bool {
```

Comment:
nit: I'm also cool with renaming `drainCtx` back to `ctx` as its contained within the function scope.

Reply:
There would be two ctx one for `Run` and one for `drainShardsAttempt`. Leaving as drainCtx as to not cause confustion.